packages/service/README.md

```diff
@@ -0,0 +1,111 @@
+# @loopback/service
+
+This module provides a common set of interfaces for interacting with service
+oriented backends such as REST APIs, SOAP Web Services, and gRPC microservices.
+
+## Installation
+
+```
+$ npm install --save @loopback/service
+```
+
+## Usage
+
+### Define a data source for the service backend
+
+```ts
+import {DataSourceConstructor, juggler} from '@loopback/service';
+
+const ds: juggler.DataSource = new DataSourceConstructor({
+  name: 'GoogleMapGeoCode',
+  connector: 'rest',
+  options: {
+    headers: {
+      'accept': 'application/json',
+      'content-type': 'application/json'
+    }
+  },
+  operations: [
+    {
+      template: {
+        method: 'GET',
+        url: 'http://maps.googleapis.com/maps/api/geocode/{format=json}',
+        query: {
+          address: '{street},{city},{zipcode}',
+          sensor: '{sensor=false}'
+        },
+        responsePath: '$.results[0].geometry.location[0]'
+      },
+      functions: {
+        geocode: ['street', 'city', 'zipcode']
+      }
+    }
+  ]
+});
+```
+
+### Bind data sources to the context
+
+```ts
+import {Context} from '@loopback/context';
+
+const context = new Context();
+context.bind('dataSources.geoService').to(ds);
+```
+
+**NOTE**: Once we start to support declarative datasources with `@loopback/boot`,
+the datasource configuration files can be dropped into `src/datasources` to be
+discovered and bound automatically.
+
+### Declare the service interface
+
+```ts
+interface GeoCode {
+  lat: number;
+  lng: number;
+}
+
+interface GeoService {
+  geocode(street: string, city: string, zipcode: string): Promise<GeoCode>;
+}
+```
+
+**NOTE**: We'll introduce tools to generate TypeScript service interfaces from
+service specifications such as OpenAPI spec.
+
+### Declare service proxies for your controller
+
+```ts
+import {serviceProxy} from '@loopback/service';
+
+export class MyController {
+
+  @serviceProxy('geoService')
+  private geoService: GeoService;
```

Comment:
Are all users required to write service interfaces manually? Shouldn't there be a type-unsafe interface allowing a simpler usage?

Something along the following lines:

```ts
// provided by us
interface GenericService {
  [method: string]: Function
}


// usage
@serviceProxy('geoService')
private geoService: GenericService;
```

Or maybe it's our intention to force LB4 users to always write the service interface, to prevent them from being lazy and losing type safety?